test/components/embedders/types/test_protocol.py

```diff
@@ -0,0 +1,36 @@
+from typing import Any, Dict
+
+from haystack.components.embedders.types.protocol import TextEmbedder
+
+
+class MockTextEmbedder:
+    def run(self, text: str, param_a: str = "default", param_b: str = "another_default") -> Dict[str, Any]:
+        """
+        Mock implementation that returns a simple embedding and metadata.
+        """
+        return {"embedding": [0.1, 0.2, 0.3], "metadata": {"text": text, "param_a": param_a, "param_b": param_b}}
+
+
+def test_protocol_implementation():
+    embedder: TextEmbedder = MockTextEmbedder()  # should not raise any type errors
+
+    result = embedder.run("test text")
+    assert isinstance(result, dict)
+    assert "embedding" in result
+    assert "metadata" in result
+    assert isinstance(result["embedding"], list)
```

Comment:
Let's check that all values are float
```suggestion
    assert isinstance(result["embedding"], list)
    assert all(isinstance(x, float) for x in result["embedding"])
```